src/lib/utils/resolvers/resolveGuildChannelPredicate.ts

```diff
@@ -12,7 +12,8 @@ export function resolveGuildChannelPredicate<TChannel extends GuildBasedChannelT
 	error: TError
 ): Result<TChannel, TError | Identifiers.ArgumentGuildChannelError> {
 	const result = resolveGuildChannel(parameter, guild);
-	if (!result.isErr()) return result;
-	if (predicate(result.unwrap())) return Result.ok(result.unwrap());
+	if (result.isErr()) return result;
```

Comment:
Can we not do an `unwrapOr` / something of that nature?

Reply:
If you're talking about the type guard, there's `isOkAnd` which checks a predicate but doesn't work with type guards. If you're talking about the `isErr` call, I don't get what could be unwrapped.

Reply:
No worries then. This is good enough.